package StringLevel2;

import java.util.Scanner;

public class StudentVoting {

    static int[] generateAges(int n) {

        int[] ages = new int[n];

        for (int i = 0; i < n; i++) {
            ages[i] = (int)(Math.random() * 90) + 10;
        }

        return ages;
    }

    static String[][] checkVotingEligibility(int[] ages) {

        String[][] result = new String[ages.length][2];

        for (int i = 0; i < ages.length; i++) {

            result[i][0] = String.valueOf(ages[i]);

            if (ages[i] < 0) {
                result[i][1] = "Cannot Vote";
            } else if (ages[i] >= 18) {
                result[i][1] = "Can Vote";
            } else {
                result[i][1] = "Cannot Vote";
            }
        }

        return result;
    }

    static void display(String[][] result) {

        System.out.println("\nStudent\tAge\tVoting Status");
        System.out.println("--------------------------------");

        for (int i = 0; i < result.length; i++) {

            System.out.println(
                    (i + 1) + "\t"
                    + result[i][0] + "\t"
                    + result[i][1]
            );
        }
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int n = 10;

        int[] ages = generateAges(n);

        String[][] result = checkVotingEligibility(ages);

        display(result);
    }
}